### DISCUSSION TITLE
Tabs: not persisting values when changing tabs

### DISCUSSION THREAD

**Discussion by mKontakis** [react-hook-form/react-hook-form#2457] [Q&A]
I want to implement parts of the form being in tabs.

Although when changing tabs, the values of the previous tab are not persisted, and the registered field is removed from the `getValues()` array.

https://codesandbox.io/s/react-hook-form-array-fields-olklx?file=/src/index.js

I tried with both controlled and uncontrolled components.

Although, I have a similar case where the tabs are in a `useFieldArray`. The tabs values do work there.

**Comment by bluebill1049** [comment]
I think you can use either `shouldUnregister` look up the doc.

or 

https://codesandbox.io/s/tabs-760h9
